### PR TITLE
[Task] Add Dinky task to better support the development and execution of FlinkSQL

### LINKED ISSUE #10444: [Feature][Task] Add Dinky task to better support the development and execution of  FlinkSQL
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar feature requirement.

### Description

Dinky is an out of the box one-stop real-time computing platform dedicated to the construction and practice of Unified Batch & Streaming and Unified Data Lake & Data Warehouse. Based on Apache Flink, Dinky provides the ability to connect many big data frameworks including OLAP and Data Lake.

1.Develop FlinkSQL on the computer and obtain the job ID from the API manual.
image
2.Add Dinky job node in workflow, and configure address and task ID.
image
3.View job scheduling status from task instances.
image
4.Query the metric information of jobs from dinky's operation and maintenance center.
image

### Use case

Add Dinky task to better support the development and execution of  FlinkSQL.

### Related issues

_No response_

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by aiwenmo**
## Purpose of the pull request
Add Dinky task to better support the development and execution of FlinkSQL.

close #10444

## Verify this pull request

This pull request is code cleanup without any test coverage.

*(or)*

This pull request is already covered by existing tests, such as *(please describe tests)*.

(or)

This change added tests and can be verified as follows:

### CODE REVIEW

**Comment by aiwenmo** [comment]
demo-get-dinky-task-id(../../../../img/tasks/demo/dinky_task_id.png)
At present, Dinky is not internationalized, and English pictures will be added later.

File: dolphinscheduler-task-plugin/dolphinscheduler-task-dinky/src/main/java/org/apache/dolphinscheduler/plugin/task/dinky/DinkyTask.java [hunk 1]
```diff
@@ -0,0 +1,272 @@
...
+                            finishFlag = true;
+                            break;
+                        case DinkyTaskConstants.STATUS_FAILED:
+                        case DinkyTaskConstants.STATUS_CANCELED:
+                        case DinkyTaskConstants.STATUS_UNKNOWN:
+                            errorHandle(jobInstanceInfoResult.get(DinkyTaskConstants.API_RESULT_DATAS).get("error").asText());
+                            finishFlag = true;
+                            break;
+                        default:
+                            Thread.sleep(DinkyTaskConstants.SLEEP_MILLIS);
```
Comment (caishunfeng) [inline dolphinscheduler-task-plugin/dolphinscheduler-task-dinky/src/main/java/org/apache/dolphinscheduler/plugin/task/dinky/DinkyTask.java]: Why to sleep here? What state does default represent?

File: dolphinscheduler-task-plugin/dolphinscheduler-task-dinky/src/main/java/org/apache/dolphinscheduler/plugin/task/dinky/DinkyTask.java [hunk 2]
```diff
@@ -0,0 +1,272 @@
...
+                    }
+                }
+            }
+        } catch (Exception e) {
+            errorHandle(e);
```
Comment (caishunfeng) [inline dolphinscheduler-task-plugin/dolphinscheduler-task-dinky/src/main/java/org/apache/dolphinscheduler/plugin/task/dinky/DinkyTask.java]: It seems unnecessary to package error handle. You can handle error directly.

File: dolphinscheduler-task-plugin/dolphinscheduler-task-dinky/src/main/java/org/apache/dolphinscheduler/plugin/task/dinky/DinkyTask.java [hunk 3]
```diff
@@ -0,0 +1,272 @@
...
+            }
+            URI uri = uriBuilder.build();
+            httpGet = new HttpGet(uri);
+            logger.info("access url: {}", uri);
+            HttpResponse response = httpClient.execute(httpGet);
+            if (response.getStatusLine().getStatusCode() == HttpStatus.SC_OK) {
+                result = EntityUtils.toString(response.getEntity());
+                logger.info("dinky task succeed with results: {}", result);
+            } else {
+                logger.info("dinky task terminated,response: {}", response);
```
Comment (caishunfeng) [inline dolphinscheduler-task-plugin/dolphinscheduler-task-dinky/src/main/java/org/apache/dolphinscheduler/plugin/task/dinky/DinkyTask.java]: Change the log level to `warn` or `error` is better, WDYT?

File: dolphinscheduler-task-plugin/dolphinscheduler-task-dinky/src/main/java/org/apache/dolphinscheduler/plugin/task/dinky/DinkyTask.java (hunk 1 above)
Comment (aiwenmo) [reply dolphinscheduler-task-plugin/dolphinscheduler-task-dinky/src/main/java/org/apache/dolphinscheduler/plugin/task/dinky/DinkyTask.java]: Hi. Need to poll for task status.

File: dolphinscheduler-task-plugin/dolphinscheduler-task-dinky/src/main/java/org/apache/dolphinscheduler/plugin/task/dinky/DinkyTask.java (hunk 2 above)
Comment (aiwenmo) [reply dolphinscheduler-task-plugin/dolphinscheduler-task-dinky/src/main/java/org/apache/dolphinscheduler/plugin/task/dinky/DinkyTask.java]: Hi. Do you mean to remove 'try catch' ?

File: dolphinscheduler-task-plugin/dolphinscheduler-task-dinky/src/main/java/org/apache/dolphinscheduler/plugin/task/dinky/DinkyTask.java (hunk 3 above)
Comment (aiwenmo) [reply dolphinscheduler-task-plugin/dolphinscheduler-task-dinky/src/main/java/org/apache/dolphinscheduler/plugin/task/dinky/DinkyTask.java]: It's a good advice. thx

**Comment by SbloodyS** [comment]
> SonarCloud Quality Gate failed.    [Quality Gate failed](https://sonarcloud.io/dashboard?id=apache-dolphinscheduler&pullRequest=10640)
> 
> [Bug](https://sonarcloud.io/project/issues?id=apache-dolphinscheduler&pullRequest=10640&resolved=false&types=BUG) [C](https://sonarcloud.io/project/issues?id=apache-dolphinscheduler&pullRequest=10640&resolved=false&types=BUG) [1 Bug](https://sonarcloud.io/project/issues?id=apache-dolphinscheduler&pullRequest=10640&resolved=false&types=BUG) [Vulnerability](https://sonarcloud.io/project/issues?id=apache-dolphinscheduler&pullRequest=10640&resolved=false&types=VULNERABILITY) [A](https://sonarcloud.io/project/issues?id=apache-dolphinscheduler&pullRequest=10640&resolved=false&types=VULNERABILITY) [0 Vulnerabilities](https://sonarcloud.io/project/issues?id=apache-dolphinscheduler&pullRequest=10640&resolved=false&types=VULNERABILITY) [Security Hotspot](https://sonarcloud.io/project/security_hotspots?id=apache-dolphinscheduler&pullRequest=10640&resolved=false&types=SECURITY_HOTSPOT) [A](https://sonarcloud.io/project/security_hotspots?id=apache-dolphinscheduler&pullRequest=10640&resolved=false&types=SECURITY_HOTSPOT) [0 Security Hotspots](https://sonarcloud.io/project/security_hotspots?id=apache-dolphinscheduler&pullRequest=10640&resolved=false&types=SECURITY_HOTSPOT) [Code Smell](https://sonarcloud.io/project/issues?id=apache-dolphinscheduler&pullRequest=10640&resolved=false&types=CODE_SMELL) [A](https://sonarcloud.io/project/issues?id=apache-dolphinscheduler&pullRequest=10640&resolved=false&types=CODE_SMELL) [0 Code Smells](https://sonarcloud.io/project/issues?id=apache-dolphinscheduler&pullRequest=10640&resolved=false&types=CODE_SMELL)
> 
> [](https://sonarcloud.io/component_measures?id=apache-dolphinscheduler&pullRequest=10640&metric=new_coverage&view=list) [0.0% Coverage](https://sonarcloud.io/component_measures?id=apache-dolphinscheduler&pullRequest=10640&metric=new_coverage&view=list) [](https://sonarcloud.io/component_measures?id=apache-dolphinscheduler&pullRequest=10640&metric=new_duplicated_lines_density&view=list) [0.0% Duplication](https://sonarcloud.io/component_measures?id=apache-dolphinscheduler&pullRequest=10640&metric=new_duplicated_lines_density&view=list)

Please fix the bugs.

File: dolphinscheduler-task-plugin/dolphinscheduler-task-dinky/src/main/java/org/apache/dolphinscheduler/plugin/task/dinky/DinkyTask.java (hunk 2 above)
Comment (caishunfeng) [reply dolphinscheduler-task-plugin/dolphinscheduler-task-dinky/src/main/java/org/apache/dolphinscheduler/plugin/task/dinky/DinkyTask.java]: > Hi. Do you mean to remove 'try catch' ?

No, I mean remove the `errorHandle` method and handle error directly.  Just a suggestion, nip.

File: dolphinscheduler-task-plugin/dolphinscheduler-task-dinky/src/main/java/org/apache/dolphinscheduler/plugin/task/dinky/DinkyTaskChannel.java [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+import org.apache.dolphinscheduler.plugin.task.api.parameters.AbstractParameters;
+import org.apache.dolphinscheduler.plugin.task.api.parameters.ParametersNode;
+import org.apache.dolphinscheduler.plugin.task.api.parameters.resource.ResourceParametersHelper;
+import org.apache.dolphinscheduler.spi.utils.JSONUtils;
+
+public class DinkyTaskChannel implements TaskChannel {
+
+    @Override
+    public void cancelApplication(boolean status) {
+        // nothing to do
```
Comment (SbloodyS) [inline dolphinscheduler-task-plugin/dolphinscheduler-task-dinky/src/main/java/org/apache/dolphinscheduler/plugin/task/dinky/DinkyTaskChannel.java:32]: Why not implement this?

**Review by SbloodyS — DISMISSED** [review]
Backend part LGTM. Thanks.

File: dolphinscheduler-ui/src/views/projects/task/components/node/tasks/use-dinky.ts [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+import {reactive} from 'vue'
+import * as Fields from '../fields/index'
+import type {IJsonItem, INodeData, ITaskData} from '../types'
+
+export function useDinky({
+                             projectCode,
+                             from = 0,
+                             readonly,
+                             data
+                         }: {
```
Comment (Amy0104) [inline dolphinscheduler-ui/src/views/projects/task/components/node/tasks/use-dinky.ts]: It's better to format the codes by running 'npm run prettier'.

File: dolphinscheduler-ui/src/views/projects/task/components/node/tasks/use-dinky.ts [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+        flag: 'YES',
+        description: '',
+        timeoutFlag: false,
+        localParams: [],
+        environmentCode: null,
+        failRetryInterval: 1,
+        failRetryTimes: 0,
+        workerGroup: 'default',
+        delayTime: 0,
+        timeout: 30
```
Comment (Amy0104) [inline dolphinscheduler-ui/src/views/projects/task/components/node/tasks/use-dinky.ts]: Here should set the timeoutNotifyStrategy default value to ['WARN'].
plx check this #10259 .

**Review by Amy0104 — CHANGES_REQUESTED** [review]

File: dolphinscheduler-ui/src/views/projects/task/components/node/tasks/use-dinky.ts (hunk 6 above)
Comment (aiwenmo) [reply dolphinscheduler-ui/src/views/projects/task/components/node/tasks/use-dinky.ts]: Hi. I have changed the code.

**Comment by aiwenmo** [comment]
Hi. Please review the new code. thx. @Amy0104

**Review by Amy0104 — DISMISSED** [review]
The front end part LGTM.

**Comment by SbloodyS** [comment]
@zhongjiajie PTAL.

File: docs/docs/en/guide/task/dinky.md [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+- Descriptive information: Describe the function of the node.
+- Task priority: When the number of worker threads is insufficient, execute in the order of priority from high to low, and tasks with the same priority will execute in a first-in first-out order.
+- Worker grouping: Assign tasks to the machines of the worker group to execute. If `Default` is selected, randomly select a worker machine for execution.
+- Number of failed retry attempts: The failure task resubmitting times. It supports drop-down and hand-filling.
+- Failed retry interval: The time interval for resubmitting the task after a failed task. It supports drop-down and hand-filling.
+- Timeout alarm: Check the timeout alarm and timeout failure. When the task exceeds the "timeout period", an alarm email will send and the task execution will fail.
+- Dinky Address: The url for a dinky server.
+- Dinky Task ID: The unique task id for a dinky task.
+- Online Task: Specify whether the current dinky job is online. If yes, the submitted job can only be submitted successfully when it is published and there is no corresponding Flink job instance running.
```
Comment (zhongjiajie) [inline docs/docs/en/guide/task/dinky.md]: currently, all our English version tasks parameters convert from list style to table format style, which you can see example in https://github.com/apache/dolphinscheduler/blob/dev/docs/docs/en/guide/task/shell.md.  So please change to a table format in English version of docs. And it is up to you whether change Chinese version of docs

**Review by zhongjiajie — COMMENTED** [review]
And, please add content in https://github.com/apache/dolphinscheduler/blob/dev/docs/configs/docsdev.js#L169-L171 and https://github.com/apache/dolphinscheduler/blob/dev/docs/configs/docsdev.js#L548-L551

**Comment by aiwenmo** [comment]
Hi. Please review the new docs. thx. @zhongjiajie

**Review by zhongjiajie — APPROVED** [review]
LGTM, thanks

**Comment by zhongjiajie** [comment]
Thanks for bringing this up

**Pull request merged**